metrics/exporter.go

```diff
@@ -111,6 +152,29 @@ func resetCurPromSrv() {
 	}
 }
 
+func resetMonitoredResourceFunc() {
+	metricsMux.Lock()
+	defer metricsMux.Unlock()
+	if getMonitoredResourceFunc != nil {
+		getMonitoredResourceFunc = nil
+	}
+}
+
+func setMonitoredResourceFunc(config *metricsConfig) {
+	metricsMux.Lock()
+	defer metricsMux.Unlock()
+	if getMonitoredResourceFunc == nil {
+		gm := retrieveGCPMetadata()
+		metricsPrefix := config.domain + "/" + config.component
+		fmt.Println("metrics prefix", metricsPrefix)
```

Comment:
We should use the proper logger here. not fmt.Println.

Reply:
Done.